src/encoding.rs

```diff
@@ -794,13 +794,15 @@ macro_rules! length_delimited {
 pub mod string {
     use super::*;
 
-    pub fn encode<B>(tag: u32, value: &String, buf: &mut B)
+    pub fn encode<S, B>(tag: u32, value: &S, buf: &mut B)
```

Comment:
I would like to see tests to guarantee `String` can still be used on this function.

And you need to add tests for your new use case, to make sure it is not broken in the future.